internal/transport/proxy_test.go

```diff
@@ -138,15 +138,17 @@ func testHTTPConnect(t *testing.T, proxyURLModify func(*url.URL) *url.URL, proxy
 	defer overwrite(hpfe)()
 
 	// Dial to proxy server.
-	dialer := newProxyDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+	oldProxyDialer := proxyDialer
+	proxyDialer = func(ctx context.Context, string, addr string) (net.Conn, error) {
```

Comment:
Are you sure we need this?  This looks equivalent to using the context dialer directly to me.  I think the only reason it was here before was because the dialer was a required parameter.

Reply:
Made this change. Got rid of the solution to a previous comment having a global dialer that was overwritten for testing because this was the only test using a different dialer. The default behavior of proxyDial does not break the test.